recipe/gen_patch_json.py

```diff
@@ -3175,6 +3175,10 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
             if pind is not None:
                 record["depends"][pind] = record["depends"][pind] + ",<3.10"
 
+        if record_name == "qcportal" and record.get("timestamp", 0) < 1691162578000:
+            # QCPortal does not work with Pydantic 2, and no released version has.
+            record["depends"].append("pydantic<2")
```

Comment:
Since `pydantic` already a dependency of `qcportal`, we prefer using `_replace_pin` instead.

[See this block for an example](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/d5473441b2742bd546bbd7053c148d0cf464d491/recipe/gen_patch_json.py#L1978-L1995).